sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -62,7 +62,7 @@ enum class HttpMethod
   HEAD,
   POST,
   PUT,
-  DELETE,
+  DELETE_TODO,
```

Comment:
This rename seems unwarranted. `DELETE` is not a keyword, and the PascalCase form we settled on doesn't step on the toes of `operator delete`.